### PR TITLE
smooth the language and update instructions in the json example doc

### PR DESCRIPTION
**Pull request by dawedawe**
I left out the inclusion of signature files on purpose as the lexer signature currently needs some manual editing to be valid.

### CODE REVIEW

File: docs/content/jsonParserExample.md [hunk 1]
```diff
@@ -266,24 +270,24 @@ You can also add ``print`` function to JsonValue and show result of parsing in c
...
-              | List values ->  values
-                                 |> List.map (fun value -> JsonValue.print(value)) 
-                                 |> String.concat ","
-                                 |> sprintf "List(%s)"
-                                 |> sprintf "List(%s)"
-
-And then change program to print result of parsing in console, add this line after line 18:
+      //below function is not important, it simply prints values 
+      static member print x = 
+                match x with
```
Comment (nojaf) [inline docs/content/jsonParserExample.md:275]: Nit: the match still looks too far indented.

**Review by nojaf — APPROVED** [review]
Great addition, thank you very much!

One thing I noticed in the existing code is:
```fsharp
{

module Lexer

open FSharp.Text.Lexing
open System
open Parser

exception SyntaxError of string

let lexeme = LexBuffer<_>.LexemeString

let newline (lexbuf: LexBuffer<_>) = 
  lexbuf.StartPos <- lexbuf.StartPos.NextLine
}

let int = ['-' '+']? ['0'-'9']+
let digit = ['0'-'9']
let frac = '.' digit*
let exp = ['e' 'E'] ['-' '+']? digit+
let float = '-'? digit* frac? exp?

let white = [' ' '\t']+
let newline = '\r' | '\n' | "\r\n"
``` 

I would have guessed that the `let int =` should also be in the `{ }` block.
Any idea if that is the case? Did you happen to run the local sample?

**Comment by dawedawe** [comment]
> I would have guessed that the `let int =` should also be in the `{ }` block. Any idea if that is the case? Did you happen to run the local sample?

Thanks for the review.
The regex macros should really go after the top code. One can see that also from https://github.com/fsprojects/FsLexYacc/blob/21bd8dbc42de7ff97a79b8502c29f1830b2c8d90/src/FsLex.Core/fslexpars.fsy#L24-L27

**Comment by nojaf** [comment]
Gotcha, thanks!

**Pull request merged**